b2share/modules/communities/cli.py

```diff
@@ -0,0 +1,120 @@
+
+#from .errors import RootSchemaAlreadyExistsError
+#from .helpers import load_root_schemas
+
+@click.group()
+def communities():
+    """communities management commands."""
+
+
+@communities.command()
+@with_appcontext
+@click.option('-v', '--verbose', is_flag=True, default=False)
+@click.argument('name')
+@click.argument('description')
+@click.argument('logo')
+def create(verbose, name, description, logo):
+    """Creating a community in the database. Name can be 255 characters long. Description is a text of maximally 1024 characters enclosed in parentheses. The logo parameter should be a valid path to a logo file. """
+    if len(name)>255:
+        raise click.BadParameter("NAME parameter is longer than the 255 character maximum")
+    if len(description)>1024:
+        raise click.BadParameter("DESCRIPTION parameter is longer than the 1024 character maximum")
+    if not isfile(logo):
+        raise click.BadParameter("LOGO should be a (relative) path to an existing image file.")
+    try:
+        community_exists = Community.get(name=name)
+        if community_exists:
+            raise click.BadParameter("There already is a community with name %s" % name)
+    except CommunityDoesNotExistError as e:
+        pass
+    try:
+        community = Community.create_community(name=name,description=description,logo=logo)
+        db.session.commit()
+        if verbose:
+            click.echo("Community created with %d" % community.id)
+    except Exception as e1:
+        try:
+            db.session.rollback()
+        except Exception as e2:
+            raise click.ClickException(str(e2))
+        raise click.ClickException(str(e1))
+        
+@communities.command()
+@with_appcontext
+@click.option('-v','--verbose', is_flag=True, default=False)
+def list(verbose):
+    """List all communities in this instances' database"""
+    communities = Community.get_all()
+    for c in communities:
+        click.echo("%s\t%s\t%s\t%s" % (c.name, c.id, c.description, c.logo))
+
+
+@communities.command()
+@with_appcontext
+@click.option('-v','--verbose',is_flag=True,default=False)
+@click.option('--name')
+@click.option('--description')
+@click.option('--logo')
+@click.option('--clear_fields',is_flag=True,default=False,help='if set edit nullifies unspecified value options')
+@click.argument('id')
+def edit(verbose, id, name, description, logo,clear_fields):
+        """Edit data of the specified community."""
+        try:
+            community = Community.get(id=id)
+        except:
+            #CommunityDoesNotExistError, ValueError, sqlalchemy.exc.StatementError
+            raise click.BadParameter("No community with id %s" % id)
+        if not(name or description or logo):
+            raise click.ClickException("At least one of name, description or id must be specified")
+        data = {}
+        if name:
+            if not(name==community.name):
+                try:
+                    Community.get(name=name)
```

Comment:
This line will get the community named by 'name', if any. It will not raise an error if the community exists, on the contrary, it will raise a CommunityDoesNotExistError.